python/ray/serve/tests/test_metrics.py

```diff
@@ -32,6 +32,21 @@
 TEST_METRICS_EXPORT_PORT = 9999
 
 
+@pytest.fixture
+def clean_up_metrics():
+    prometheus_base_url = f"http://127.0.0.1:{TEST_METRICS_EXPORT_PORT}"
+    delete_all_series_url = (
+        f"{prometheus_base_url}/api/v1/admin/tsdb"
+        '/delete_series?match[]={__name__=~".*"}'
+    )
+    clean_tombstones_url = f"{prometheus_base_url}/api/v1/admin/tsdb/clean_tombstones"
+    requests.post(delete_all_series_url)
+    requests.post(clean_tombstones_url)
+    yield
+    requests.post(delete_all_series_url)
+    requests.post(clean_tombstones_url)
+
+
 @pytest.fixture
 def serve_start_shutdown():
     """Fixture provides a fresh Ray cluster to prevent metrics state sharing."""
```

Comment:
Hm shouldn't this be sufficient on its own? The prometheus endpoint is the raylet, so if ray is shut down between runs there should be no state sharing.

What am I missing?

Reply:
My expectation is there are something that's not cleaned up in between those tests. And in fact adding those calls seems to helped. Now that thinking through it again maybe just adding some sleep in between will also help the same way and maybe the issue is serve and/or ray wasn't complete shutdown before the next test starts? 🤔 Let me do some more experiments

Reply:
we should not add any sleeps -- if we need to wait for anything to clean up, then explicitly wait for the cleanup to happen

sleeps are what make things flaky in the first place